lua/xbase/treesitter.lua

```diff
@@ -0,0 +1,173 @@
+local have_treesitter = pcall(require, "nvim-treesitter")
+if not have_treesitter then
+    return nil
+end
+
+local ts_utils = require("nvim-treesitter.ts_utils")
+local get_node_text = function(node)
+	return vim.treesitter.get_node_text(node, vim.api.nvim_get_current_buf())
+end
+local get_winr = vim.api.nvim_get_current_win
+local get_current_node = ts_utils.get_node_at_cursor
+
+local property_declaration_query = vim.treesitter.query.parse_query("swift", "(property_declaration) @prop")
```

Comment:
Okay now this need to be wrapped in a pcall, because user might have swift installed. same thing return nil if doesn't exists

Reply:
Maybe a requirement for swift tree-sitter be installed needed to be listed

Reply:
Would it also make sense to send a notification asking the user to install the swift grammer with treesitter?

Reply:
I'd avoid that at the time being because xbase right now attach at startup on nvim regardless whether it's in swift/xbase project or not. 

Create an issue for that I will later resolve after making xbase loading tided to to swift/xcode projects